src/components/Events/index.js

```diff
@@ -4,25 +4,125 @@
  * file, You can obtain one at https://mozilla.org/MPL/2.0/.
  */
 
-import React, { useMemo } from 'react';
+import React, { useEffect, useMemo, useRef, useState } from 'react';
 import PropTypes from 'prop-types';
 
 import ReadMore from '../ReadMore';
 import Timeline from './components/Timeline';
+import Dropdown from '../Dropdown';
 
 import { aggregateCountOfEventProperty } from './lib';
 
 import './styles.css';
 
-const Events = ({ events }) => {
+const filterValues = [
+  "Last 15 minutes",
+  "Last 30 minutes",
+  "Last 1 hour",
+  "Last 3 hours",
+  "Last 6 hours",
+  "Last 12 hours",
+  "Last 1 day",
+  "Last 7 days",
+  "Last 21 days"
+];
+
+const Events = ({ events, header, isEventStream }) => {
+  const listRefs = useRef([]);
+
+  /// state ///
+  const [filterValue, setFilterValue] = useState("");
+  const [lastFilterHadNoEvents, setLastFilterHadNoEvents] = useState(undefined);
+  const [scrollingKey, setScrollingKey] = useState("");
+
+  /// helpers ///
+  const filterEvents = (eventsToFilter, filterDate) => {
+    return eventsToFilter.filter((event) => event.timestamp > filterDate.getTime()) || [];
+  };
+
   const trimmedEvents = useMemo(() => {
+    let normalizedEvents;
+
     // Event limit is 500, anything more than that will be truncated.
     if (events && events.length && events.length > 500) {
-      return events.slice(0, 500);
+      normalizedEvents = events.slice(0, 500);
```

Comment:
Why do we truncate to 500 events here?

Reply:
Anything more than 500 events is likely going to have some sort of performance implication. Even at 500 you can feel things start to get the smallest bit sluggish. If we have people complain about this we can look to optimize things so that 500 works out, but I am hoping for now this shouldn't be an issue.